### PR TITLE
JDBetteridge/petsc testing

### PR DESCRIPTION
**Pull request by JDBetteridge**
# Description
DO NOT MERGE
Test removing:
- chaco
- superlu_dist
- mumps and scalapack
- hypre
- others...

### CODE REVIEW

File: firedrake/petsc.py [hunk 1]
```diff
@@ -334,3 +370,70 @@ def garbage_view(obj: Any):
...
+# > ICNTL(14) controls the percentage increase in the estimated working space...
+# > ... Remarks: When significant extra fill-in is caused by numerical pivoting, increasing ICNTL(14) may help.
+if DEFAULT_DIRECT_SOLVER == "mumps":
+    DEFAULT_DIRECT_SOLVER_PARAMETERS["mat_mumps_icntl_14"] = 200
+
+# Setup default AMG preconditioner
+amg_priority = ["hypre", "ml"]
+for amg in amg_priority:
+    if amg in external_packages:
+        DEFAULT_AMG_PC = amg
```
Comment (connorjward) [inline firedrake/petsc.py:419]: missing a `break`.

File: firedrake/petsc.py [hunk 2]
```diff
@@ -334,3 +370,70 @@ def garbage_view(obj: Any):
...
+    )
+    DEFAULT_DIRECT_SOLVER = "petsc"
+    DEFAULT_DIRECT_SOLVER_PARAMETERS = {"mat_solver_type": "petsc"}
+
+# MUMPS needs an additional parameter set
+# From the MUMPS documentation:
+# > ICNTL(14) controls the percentage increase in the estimated working space...
+# > ... Remarks: When significant extra fill-in is caused by numerical pivoting, increasing ICNTL(14) may help.
+if DEFAULT_DIRECT_SOLVER == "mumps":
+    DEFAULT_DIRECT_SOLVER_PARAMETERS["mat_mumps_icntl_14"] = 200
```
Comment (connorjward) [inline firedrake/petsc.py:413]: I don't really like this pattern. We have constants here that aren't really constants. I would probably prefer this to be called `DEFAULT_MUMPS_SOLVER_PARAMETERS`. That or make this a function `get_default_direct_solver_options("mumps")`.

I also think that if users are advanced enough to be playing with solver options then we shouldn't obfuscate things behind ambiguous variables. Perhaps this sort of thing could be put into `conftest.py` as a fixture so we get the right behaviour for the tests but not touch the user experience?

File: firedrake/petsc.py (hunk 1 above)
Comment (JDBetteridge) [reply firedrake/petsc.py:419]: This is the wrong PR see https://github.com/firedrakeproject/firedrake/pull/3348

File: firedrake/petsc.py (hunk 2 above)
Comment (JDBetteridge) [reply firedrake/petsc.py:413]: This is the wrong PR see https://github.com/firedrakeproject/firedrake/pull/3348

**Comment by JDBetteridge** [comment]
This PR is out of date and is marked DO NOT MERGE

**Pull request closed without merging**